test/utils/data.test.js

```diff
@@ -1229,35 +1238,78 @@ describe('DataUtil', () => {
     });
   });
 
-  describe('getTotalInsulinData', () => {
-    it('should return the total basal and bolus insulin delivery when viewing 1 day', () => {
-      dataUtil.endpoints = dayEndpoints;
-      expect(dataUtil.getTotalInsulinData()).to.eql({
+  describe('getTotalInsulinAndWeightData', () => {
+    it('should return the total basal and bolus insulin delivery when viewing 1 day and weight', () => {
+      let du = new DataUtil(data, defaultOpts);
+      du.endpoints = dayEndpoints;
+      du.addData(pumpSettings);
+      // {name: 'WEIGHT', value: '60', unit: 'kg', level: 1}
+      expect(du.getTotalInsulinAndWeightData()).to.eql({
         totalInsulin: 16.5,
+        weight: {
+          name: 'WEIGHT',
+          value: '60',
+          unit: 'kg',
+          level: 1
+        }
       });
     });
 
-    it('should return the avg daily total basal and bolus insulin delivery when viewing more than 1 day', () => {
-      dataUtil.endpoints = twoWeekEndpoints;
-      expect(dataUtil.getTotalInsulinData()).to.eql({
+    it('should return the avg daily total basal and bolus insulin delivery when viewing more than 1 day and weight', () => {
+      let du = new DataUtil(data, defaultOpts);
+      du.endpoints = twoWeekEndpoints;
+      du.addData(pumpSettings);
+      expect(du.getTotalInsulinAndWeightData()).to.eql({
         totalInsulin: 10.5, // 9.5 + 1
+        weight: {
+          name: 'WEIGHT',
+          value: '60',
+          unit: 'kg',
+          level: 1
+        }
+      });
+    });
+
+    it('should return undefined weight when no pump settings found', () => {
+      let du = new DataUtil(data, defaultOpts);
+      du.endpoints = dayEndpoints;
+      console.log(du.getTotalInsulinAndWeightData());
+      expect(du.getTotalInsulinAndWeightData()).to.eql({
+        totalInsulin: 16.5,
+        weight: undefined
```

Comment:
```suggestion
    it('should return null weight when no pump settings found', () => {
      const du = new DataUtil(data, defaultOpts);
      du.endpoints = dayEndpoints;
      expect(du.getTotalInsulinAndWeightData()).to.eql({
        totalInsulin: 16.5,
        weight: null,
      });
```
To fix the test if previous suggestions are applied.